crates/esthri-cli/src/cli_utils.rs

```diff
@@ -136,3 +138,40 @@ pub fn setup_upload_termination_handler(s3: S3Client) {
         std::process::exit(0);
     });
 }
+
+pub fn setup_s3client_with_cred_provider() -> S3Client {
+    let mut hyper_builder = Client::builder();
+    hyper_builder.pool_idle_timeout(Duration::from_secs(20));
+
+    let https_connector = esthri::new_https_connector();
+    let http_client = HttpClient::from_builder(hyper_builder, https_connector);
+
+    match std::env::var("ESTHRI_CREDENTIAL_PROVIDER") {
+        Ok(val) => match val.as_str() {
+            "env" => {
+                let credentials_provider = EnvironmentProvider::default();
+                S3Client::new_with(http_client, credentials_provider, Region::default())
+            }
+            "profile" => {
+                let credentials_provider = ProfileProvider::new().unwrap();
+                S3Client::new_with(http_client, credentials_provider, Region::default())
+            }
+            "container" => {
+                let credentials_provider = ContainerProvider::new();
+                S3Client::new_with(http_client, credentials_provider, Region::default())
+            }
+            "instance_metadata" => {
+                let credentials_provider = InstanceMetadataProvider::new();
+                S3Client::new_with(http_client, credentials_provider, Region::default())
+            }
+            _ => {
```

Comment:
As a quality of life improvement we should allow using `ESTHRI_CREDENTIAL_PROVIDER= esthri .,.` to mean the default credential provider, this will allow users to easily override an environment variable without have to delete it from the environment.